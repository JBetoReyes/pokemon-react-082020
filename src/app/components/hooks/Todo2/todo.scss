.list-group-item {
  display: flex;
  align-items: center;
  p {
    min-width: 200px;
    cursor: pointer;
    &.todo-item--completed {
      text-decoration: line-through;
    }
  }
}
