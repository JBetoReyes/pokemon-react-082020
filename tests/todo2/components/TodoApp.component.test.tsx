import TodoApp from '@components/hooks/Todo2/TodoApp.component';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from '@testing-library/react';
import { ITodoApp } from '@components/hooks/Todo2/todo.model';
import demoTodos2 from '../../fixtures/demoTodos2';

describe('TodoApp', () => {
  const wrapper = shallow(<TodoApp />);
  test('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should add two items', () => {
    const todoComp = mount(<TodoApp />);
    act(() => {
      const handleAdd = todoComp.find('TodoAdd').prop('handleAdd') as (
        newTodo: ITodoApp
      ) => void;
      handleAdd(demoTodos2[0]);
      handleAdd(demoTodos2[1]);
    });
    expect(todoComp.find('h1').text().trim()).toBe('Todo App (2)');
  });
});
