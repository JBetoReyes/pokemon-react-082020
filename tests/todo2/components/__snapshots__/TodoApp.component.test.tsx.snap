// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp Should match the snapshot 1`] = `
<Fragment>
  <h1
    className="m-3"
  >
    Todo App (0)
  </h1>
  <hr />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <TodoList
          handleDelete={[Function]}
          handleTodo={[Function]}
          todos={Array []}
        />
      </div>
      <div
        className="col"
      >
        <TodoAdd
          handleAdd={[Function]}
        />
      </div>
    </div>
  </div>
</Fragment>
`;
